examples/chip-tool/main.cpp

```diff
@@ -33,11 +33,42 @@
 
 #include <zap-generated/cluster/Commands.h>
 
+#include "platform/PlatformManager.h"
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+using namespace chip::WiFiPAF;
+#endif
+
+#if CHIP_DEVICE_LAYER_TARGET_LINUX && CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+static void StopSignalHandler(int signum)
```

Comment:
So a larger question:  Shouldn't this just happen via the normal shutdown codepath, instead of directly off the signal handler?  As in, is the issue just that chip-tool does not do a "clean shutdown" on Ctrl-C?  But then how is the "just run to completion" handled?

Basically: I don't understand why we are doing this signal handler at all, and only doing it for chip-tool, not server apps.

Reply:
There is no issue in the normal execution path that CHIPCommand::MaybeTearDownStack() is called to shutdown the commissioner properly. The PAF shutdown function is invoked as part of the process.

However, a problem arises when the user terminate the chip-tool application abnormally using ctrl+c. Since no signal handling function is hooked, the program exits immediately and the shutdown function is not called.

The other apps (like all-clusters-app on Linux), the signal handling function is implemented in example/linux/AppMain.cpp. But it's no included in chip-tool project